### ISSUE TITLE
SC1073: version: 0.3.5: bug with parsing if inside case

### ISSUE THREAD

**Issue by franzinc** [koalaman/shellcheck#925]
This still fails on shellcheck.net.

```sh

#! /bin/bash

distrule=$1
installdcls=$2

case $distrule in
    alldist)
	die '--distrule alldist is not a good idea (see bug15923)'
	;;
    cleancleandist|cleandist|dist|qdist)
	if [ "$installdcls" -a '(' "$distrule" = "qdist" ')' ]; then 
	    die 'dcl installation is incompatible with --distrule qdist'
	fi
	;;
    *) die invalid dist rule: $distrule ;;
esac

```

#### Here's what shellcheck currently says:

```
In foo.sh line 11:
	if [ "$installdcls" -a '(' "$distrule" = "qdist" ')' ]; then 
        ^-- SC1009: The mentioned parser error was in this if expression.
           ^-- SC1073: Couldn't parse this test expression.
                                   ^-- SC1072: Expected "]". Fix any mentioned problems and try again.
```

**Comment by franzinc** [comment]
Even smaller test case:

```sh

#! /bin/bash

distrule=$1
installdcls=$2

case $distrule in
    foo)
	if [ '(' "$distrule" = "qdist" ')' ]; then 
	    echo foo
	    exit 1
	fi
	;;
esac
```

**Comment by Nightfirecat** [comment]
Is this a bug? The following also gives the same errors:

```
#!/bin/bash

if [ '(' "test1" = "test2" ')' ]; then
    echo foo
    exit 1
fi
```

**Comment by koalaman** [comment]
This is a bug. `\(` is recognized but `'('` is not.

The parsing used for `[` is different from other commands to allow for more helpful error messages regarding `>` and such.

**Comment by koalaman** [comment]
It now works for typical quoted cases like `"("` and `'('`.  Thanks!
